### ISSUE TITLE
Implement `TooManyBranchesViolation`

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#24]
When code has a lot of `if`, `try`, `for`, and other branches it is hard to follow what is going on.

We can restrict the number of logic branches allowed for a single function.
That's how it should work:
- `if` is counted as `+1` for the complexity, `else` and `elif` also counts as `+1`
- `for` is counted as `+1`, `else` is also counted as `+1`
- `while` is counted as `+1`
- `try` is counted as `+1`, each `except` case is counted as `+1`
- `with` is counted as `+1`

We need to figure out what number of branches is the most pleasant to have. We can consult `pylint` for this.

**Comment by malinoff** [comment]
https://github.com/PyCQA/mccabe

**Comment by sobolevn** [comment]
Hm, seems like `mccabe` covers this already. Thanks!
